### PR TITLE
Fixes for LLVM trunk.

### PR DESCRIPTION
**Pull request by jiapei100**
…for the compatibility of LLVM 10.0.0

### CODE REVIEW

**Comment by dbabokin** [comment]
Please apply ``clang-format -i src/ctx.cpp``

**Comment by DeepakRajendrakumaran** [comment]
Hi. Quick question: Looking at the LLVM API, I see that it makes sense to use valueOrNone() when there is a chance if undefined. Did you run into any cases where this was an issue/did you have any failures due to this?

**Comment by dbabokin** [comment]
I'm not sure if adding more commits was intentional for this PR. I suggest not to expand the scope of this PR and keep it for llvm::MaybeAlign issue only. You'll need to updated your branch for this.

Original changes look good to me, but would be interesting to know what triggered these changes. @jiapei100 have you checked alignment info and was wrong?

**Comment by dbabokin** [comment]
@DeepakRajendrakumaran thanks for looking at this and amending with missing fixes to make trunk work again.

File: src/llvmutil.h [hunk 1]
```diff
@@ -51,6 +51,12 @@ class PHINode;
 class InsertElementInst;
 } // namespace llvm
 
+#if ISPC_LLVM_VERSION >= ISPC_LLVM_11_0
+#define LLVMVECTOR llvm::FixedVectorType
```
Comment (dbabokin) [inline src/llvmutil.h:55]: It's probably better to have type alias instead of a macro here.

**Pull request merged**